Ext4Pkg/Ext4Dxe/BlockMap.c

```diff
@@ -239,7 +239,7 @@ Ext4GetBlocks (
 
   if (BlockPathLength == 1) {
     // Fast path for blocks 0 - 12 that skips allocations
-    Ext4GetExtentInBlockMap (Inode->i_data, EXT4_DBLOCKS, BlockPath[0], Extent);
+    Ext4GetExtentInBlockMap (Inode->i_data, EXT4_DBLOCKS, (UINT32) BlockPath[0], Extent);

```

Comment:
I *really* do not like these caller casts. Can we make `Ext4GetBlockPath()` return an `UINT32` array? This is safe because `LogicalBlock` already is `UINT32` and all array member writes are `<= LogicalBlock`.

Reply:
Cc @heatd

Reply:
We can, because Blockmap is all about ext2/3 which is 32-bit, so EXT4_BLOCK_NR here is UINT32.